### DISCUSSION TITLE
How do I set X-Forwarded-For ?

### DISCUSSION THREAD

**Discussion by ghost** [NginxProxyManager/nginx-proxy-manager#2916] [Q&A] [answered]
Hi,

I am running plausible webanalytics on a docker container, behind **nginx-proxy-manager** as reverse proxy. 

To get the IP of the clients forwarder to plausible, I need to set `X-Forwarded-For` header:
https://plausible.io/docs/self-hosting#existing-reverse-proxy 

So far I have  added _a custom location '/' and add the header in the custom config_. :
`proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;`
but it does not seem to work.

How do I proceed ?  

Thanks

**Comment by amjkwd** [comment]
Because when you set up a custom / location, it will auto set these option. 

`root@nginx-1:/data/ProxyManager/data# vim nginx/proxy_host/43.conf 

  location / {
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-Scheme $scheme;
    proxy_set_header X-Forwarded-Proto  $scheme;
    proxy_set_header X-Forwarded-For    $remote_addr;
    proxy_set_header X-Real-IP          $remote_addr;`

And yours X-Forwarded-For had been coverd. I'm looking for a solution, too.

**Reply by ghost** [reply, marked as answer]
Thanks. My docker container runs on a Synology NAS. It might be that the issue is not linked to Nginx Proxy Manager.

I am now using directly Nginx inside a Docker container and have followed this tip https://www.pedrolamas.com/2020/11/04/exposing-the-client-ips-to-docker-containers-on-synology-nas/ . It works fine now. 

It would probably also work fine with Nginx Proxy Manager.

**Reply by kenaddams42** [reply]
Hi ! Have you found a solution ? Thanks.

**Reply by amjkwd** [reply]
Yeah, I had work it out. You can write your own location code in advanced tab(not custom locations tab), and it will cover the default location option.

**Reply by ghost** [reply]
Thanks for fixing that.

**Comment by john-clark** [comment]
@amjkwd What was the specific information needed in the advanced tab?

**Reply by amjkwd** [reply]
What you write in the advanced tab, will be add to nginx's server context.
So you can write just a simple option that works in server context.
Such as:

`proxy_connect_timeout 60s;`

Or, you can also add full location context into advanced tab.
Such as :

`location / {
proxy_set_header Host $host;
proxy_set_header X-Forwarded-Scheme $scheme;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-For $remote_addr;
proxy_set_header X-Real-IP $remote_addr;
}`

Try yourself and see things really changed in the conf file.

**Reply by john-clark** [reply]
This did not work at all. That line ended in 404 not found. Without the the "location / { }" and just the proxy_set_* lines Nextcloud says proxy header is incorrect. 

Edit: I believe this is because I am hitting the proxy from inside my network. When use an outside IP it works as expected.

**Reply by amjkwd** [reply]
My code in location / { } is just a exsample snnipet and only include some proxy header options only, you need orther options to make it proxy your website correctly. Google "proxy website with nginx" may help you.
